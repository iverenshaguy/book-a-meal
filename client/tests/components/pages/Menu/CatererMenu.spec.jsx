import React from 'react';
import moment from 'moment';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import CatererMenuComponent from '../../../../src/components/pages/Menu/CatererMenu';
import CatererMenuContainer from '../../../../src/containers/pages/Menu/CatererMenu';
import {
  caterer, mealsObj, initialState, metadata
} from '../../../setup/mockData';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({
  ...initialState,
  menu: { ...initialState.menu, meals: mealsObj.meals },
  auth: { ...initialState.auth, user: { ...initialState.auth.user, ...caterer } }
});
const { now } = Date;

describe('CatererMenu', () => {
  beforeAll(() => {
    Date.now = jest.fn(() => 0);
  });

  afterAll(() => {
    Date.now = now;
  });

  it('should render Menu component correctly and show Meal Cards', () => {
    const toggleMock = jest.fn();

    const shallowWrapper = shallow(<CatererMenuComponent
      user={caterer}
      fetchMenu={jest.fn()}
      {...mealsObj}
      submitting={false}
      setCurrentDay={jest.fn()}
      currentDay={moment().format('YYYY-MM-DD')}
      toggleModal={toggleMock}
      metadata={metadata}
    />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
    expect(shallowWrapper.find('MealCard')).toBeTruthy();
  });

  it('should start fetching more menu items when Menu Component triggers loadMoreMenu function', () => {
    const toggleMock = jest.fn();
    const fetchMenuMock = jest.fn();

    const shallowWrapper = shallow(<CatererMenuComponent
      user={caterer}
      fetchMenu={fetchMenuMock}
      {...mealsObj}
      submitting={false}
      setCurrentDay={jest.fn()}
      currentDay={moment().format('YYYY-MM-DD')}
      toggleModal={toggleMock}
      metadata={metadata}
    />);

    shallowWrapper.instance().loadMoreMenu();

    expect(fetchMenuMock).toHaveBeenCalledTimes(2);
  });

  it('should open Menu modal when "Set Menu" button is clicked', () => {
    const toggleMock = jest.fn();
    const wrapper = shallow(<CatererMenuComponent
      user={caterer}
      fetchMenu={jest.fn()}
      {...mealsObj}
      submitting={false}
      setCurrentDay={jest.fn()}
      currentDay={moment().format('YYYY-MM-DD')}
      toggleModal={toggleMock}
      metadata={metadata}
    />);

    wrapper.find('#menu-modal-btn').simulate('click');
    expect(toggleMock).toHaveBeenCalled();
  });

  it('should call setCurrentDay function to set the current day when "Change Date" button is clicked', () => {
    const setCurrentDayMock = jest.fn();
    const wrapper = shallow(<CatererMenuComponent
      fetchMenu={jest.fn()}
      {...mealsObj}
      submitting={false}
      setCurrentDay={setCurrentDayMock}
      currentDay={moment().format('YYYY-MM-DD')}
      toggleModal={jest.fn()}
      metadata={metadata}
    />);

    const event = { target: { value: moment().format('YYYY-MM-DD') } };

    wrapper.find('DatePicker').at(0).dive().find('input')
      .simulate('change', event);
    expect(setCurrentDayMock).toHaveBeenCalled();
  });

  it('should render message when there are no meals on the menu', () => {
    const toggleMock = jest.fn();
    const shallowWrapper = shallow(<CatererMenuComponent
      fetchMenu={jest.fn()}
      meals={[]}
      submitting={false}
      setCurrentDay={jest.fn()}
      currentDay={moment().format('YYYY-MM-DD')}
      toggleModal={toggleMock}
      metadata={metadata}
    />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
    expect(shallowWrapper.find('p').text()).toEqual('There are no Meal Items on this Menu');
  });

  it('should not show "Set Menu" button when the time is changed to a time that is before the current time', () => {
    const toggleMock = jest.fn();
    const wrapper = shallow(<CatererMenuComponent
      fetchMenu={jest.fn()}
      {...mealsObj}
      submitting={false}
      setCurrentDay={jest.fn()}
      currentDay={moment().format('YYYY-MM-DD')}
      toggleModal={toggleMock}
      metadata={metadata}
    />);

    wrapper.setProps({
      currentDay: '1969-04-27'
    });

    expect(wrapper.find('#menu-modal-btn').length).toBeFalsy();
  });

  it('should render Menu Component when it is connected to the redux store', () => {
    const comp = (
      <Provider store={store}>
        <CatererMenuContainer
          {...mealsObj}
        />
      </Provider>
    );

    const wrapper = mount(comp, rrcMock.get());

    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
