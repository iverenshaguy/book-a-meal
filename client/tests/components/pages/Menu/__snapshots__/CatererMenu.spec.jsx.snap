// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatererMenu should render Menu Component when it is connected to the redux store 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(CatererMenu)
    meals={
      Array [
        Object {
          "description": "Jollof Rice, Beef and Plantain. 2 pieces of beef per plate",
          "id": "81211c24-51c0-46ec-b1e0-18db55880958",
          "imageUrl": "http://www.preciouscore.com/wp-content/uploads/2017/11/How-to-cook-jollof-rice-in-the-oven-750x500.jpg",
          "price": "1500.00",
          "title": "Jollof Rice, Beef and Plantain",
          "vegetarian": false,
        },
        Object {
          "description": "Sharwama contains no animal products, perfect for dieters",
          "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
          "imageUrl": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
          "price": "1200.00",
          "title": "Vegetable Sharwama and Guava Smoothie",
          "vegetarian": true,
        },
      ]
    }
  >
    <CatererMenu
      currentDay="1970-01-01"
      fetchMenu={[Function]}
      meals={
        Array [
          Object {
            "description": "Jollof Rice, Beef and Plantain. 2 pieces of beef per plate",
            "id": "81211c24-51c0-46ec-b1e0-18db55880958",
            "imageUrl": "http://www.preciouscore.com/wp-content/uploads/2017/11/How-to-cook-jollof-rice-in-the-oven-750x500.jpg",
            "price": "1500.00",
            "title": "Jollof Rice, Beef and Plantain",
            "vegetarian": false,
          },
          Object {
            "description": "Sharwama contains no animal products, perfect for dieters",
            "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
            "imageUrl": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
            "price": "1200.00",
            "title": "Vegetable Sharwama and Guava Smoothie",
            "vegetarian": true,
          },
        ]
      }
      metadata={Object {}}
      setCurrentDay={[Function]}
      toggleModal={[Function]}
    >
      <Connect(View)
        showTime={false}
        type="menu"
        updateCurrentDate={[Function]}
      >
        <View
          isFetching={false}
          logout={[Function]}
          showTime={false}
          type="menu"
          updateCurrentDate={[Function]}
          uploading={false}
          user={
            Object {
              "address": "$, Herbert Macaulay Way, Yaba, Lagos",
              "businessName": "Food Circle",
              "email": "food@circle.com",
              "firstname": "Dave",
              "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
              "lastname": "Smith",
              "phoneNo": "08166557788",
              "role": "caterer",
            }
          }
        >
          <div
            className="admin"
          >
            <Header
              active="menu"
              dateType="menu"
              logout={[Function]}
              showTime={false}
              type="caterer"
              updateCurrentDate={[Function]}
              user={
                Object {
                  "address": "$, Herbert Macaulay Way, Yaba, Lagos",
                  "businessName": "Food Circle",
                  "email": "food@circle.com",
                  "firstname": "Dave",
                  "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
                  "lastname": "Smith",
                  "phoneNo": "08166557788",
                  "role": "caterer",
                }
              }
            >
              <div
                className="header"
              >
                <div
                  className="navbar"
                >
                  <Connect(CatererHeader)
                    dateType="menu"
                    showTime={false}
                    updateCurrentDate={[Function]}
                  >
                    <CatererHeader
                      currentDay="1970-01-01"
                      dateType="menu"
                      showTime={false}
                      toggleSideNav={[Function]}
                      updateCurrentDate={[Function]}
                    >
                      <div
                        className="menu-control d-none-md"
                      >
                        <Link
                          className="d-none-md"
                          clickHandler={[Function]}
                          dataTip={null}
                          id={null}
                        >
                          <button
                            className="link-btn d-none-md"
                            data-tip={null}
                            id={null}
                            onClick={[Function]}
                            type="button"
                          >
                            <img
                              alt="toggler"
                              src="/img/toggler.png"
                            />
                          </button>
                        </Link>
                      </div>
                      <div
                        className="page-title d-none-md"
                      >
                        <h3>
                          <Link
                            replace={false}
                            to="/"
                          >
                            <a
                              href="/"
                              onClick={[Function]}
                            >
                              BOOK-A-MEAL
                            </a>
                          </Link>
                        </h3>
                      </div>
                      <div
                        className="page-title d-flex-md"
                      >
                        <div
                          className="date"
                        >
                          <h2>
                            Thursday, 1st January 1970
                             
                            <span
                              style={
                                Object {
                                  "cursor": "pointer",
                                }
                              }
                            >
                               ▾
                            </span>
                          </h2>
                          <DatePicker
                            handleSelectDate={[Function]}
                            screenSize="md"
                          >
                            <div
                              className="datepicker-div d-flex-md"
                            >
                              <input
                                id="datepicker"
                                onChange={[Function]}
                                type="date"
                              />
                            </div>
                          </DatePicker>
                        </div>
                      </div>
                    </CatererHeader>
                  </Connect(CatererHeader)>
                </div>
              </div>
            </Header>
            <div
              className="content"
            >
              <Connect(SideNav)
                active="menu"
                logout={[Function]}
                user={
                  Object {
                    "address": "$, Herbert Macaulay Way, Yaba, Lagos",
                    "businessName": "Food Circle",
                    "email": "food@circle.com",
                    "firstname": "Dave",
                    "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
                    "lastname": "Smith",
                    "phoneNo": "08166557788",
                    "role": "caterer",
                  }
                }
              >
                <SideNav
                  active="menu"
                  logout={[Function]}
                  open={false}
                  push={[Function]}
                  toggleSideNav={[Function]}
                  user={
                    Object {
                      "address": "$, Herbert Macaulay Way, Yaba, Lagos",
                      "businessName": "Food Circle",
                      "email": "food@circle.com",
                      "firstname": "Dave",
                      "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
                      "lastname": "Smith",
                      "phoneNo": "08166557788",
                      "role": "caterer",
                    }
                  }
                >
                  <div
                    className="sidenav null"
                    id="sidenav"
                  >
                    <div
                      className="sidenav-content"
                    >
                      <div
                        className="sidenav-header"
                      >
                        <CloseIcon
                          btnID={null}
                          clickHandler={[Function]}
                          divClass="d-none-md"
                        >
                          <div
                            className="close d-none-md"
                          >
                            <Link
                              aria-hidden="true"
                              className={null}
                              clickHandler={[Function]}
                              dataTip={null}
                              id={null}
                            >
                              <button
                                className="link-btn null"
                                data-tip={null}
                                id={null}
                                onClick={[Function]}
                                type="button"
                              >
                                ×
                              </button>
                            </Link>
                          </div>
                        </CloseIcon>
                        <div
                          className="sidenav-title"
                        >
                          <h3
                            className="d-flex-md"
                          >
                            <Link
                              href="/"
                              replace={false}
                              to="/"
                            >
                              <a
                                href="/"
                                onClick={[Function]}
                              >
                                BOOK-A-MEAL
                              </a>
                            </Link>
                          </h3>
                          <div
                            className="username-circle"
                          >
                            <p>
                              F
                            </p>
                          </div>
                          <p>
                            Food Circle
                          </p>
                          <span>
                            <Link
                              className={null}
                              clickHandler={[Function]}
                              dataTip={null}
                              id={null}
                            >
                              <button
                                className="link-btn null"
                                data-tip={null}
                                id={null}
                                onClick={[Function]}
                                type="button"
                              >
                                Logout
                              </button>
                            </Link>
                          </span>
                        </div>
                      </div>
                      <div
                        className="sidenav-body"
                      >
                        <Link
                          className="menu-item false"
                          clickHandler={[Function]}
                          dataTip={null}
                          id={null}
                        >
                          <button
                            className="link-btn menu-item false"
                            data-tip={null}
                            id={null}
                            onClick={[Function]}
                            type="button"
                          >
                            Dashboard
                          </button>
                        </Link>
                        <Link
                          className="menu-item false"
                          clickHandler={[Function]}
                          dataTip={null}
                          id={null}
                        >
                          <button
                            className="link-btn menu-item false"
                            data-tip={null}
                            id={null}
                            onClick={[Function]}
                            type="button"
                          >
                            Meals
                          </button>
                        </Link>
                        <Link
                          className="menu-item active"
                          clickHandler={[Function]}
                          dataTip={null}
                          id={null}
                        >
                          <button
                            className="link-btn menu-item active"
                            data-tip={null}
                            id={null}
                            onClick={[Function]}
                            type="button"
                          >
                            Menu
                          </button>
                        </Link>
                        <Link
                          className="menu-item false"
                          clickHandler={[Function]}
                          dataTip={null}
                          id={null}
                        >
                          <button
                            className="link-btn menu-item false"
                            data-tip={null}
                            id={null}
                            onClick={[Function]}
                            type="button"
                          >
                            Orders
                          </button>
                        </Link>
                      </div>
                    </div>
                  </div>
                </SideNav>
              </Connect(SideNav)>
              <div
                className="content-wrapper"
                id="has-modal"
              >
                <div
                  className="content-wrapper meals menu-meals"
                >
                  <div
                    className="date d-none-md"
                  >
                    <h2>
                      Thursday, 1st January 1970
                       
                      <span
                        style={
                          Object {
                            "cursor": "pointer",
                          }
                        }
                      >
                         ▾
                      </span>
                    </h2>
                    <DatePicker
                      handleSelectDate={[Function]}
                      screenSize={null}
                    >
                      <div
                        className="datepicker-div"
                      >
                        <input
                          id="datepicker"
                          onChange={[Function]}
                          type="date"
                        />
                      </div>
                    </DatePicker>
                  </div>
                  <div
                    className="top"
                  >
                    <button
                      className="btn btn-pri"
                      id="menu-modal-btn"
                      onClick={[Function]}
                      type="button"
                    >
                      Set Menu
                    </button>
                  </div>
                  <div
                    className="page-heading"
                  >
                    <h2>
                      Meal Items on This Day's Menu
                    </h2>
                    <hr />
                  </div>
                  <CardGroup
                    items={
                      Array [
                        <MealCard
                          inBasket={false}
                          meal={
                            Object {
                              "description": "Jollof Rice, Beef and Plantain. 2 pieces of beef per plate",
                              "id": "81211c24-51c0-46ec-b1e0-18db55880958",
                              "imageUrl": "http://www.preciouscore.com/wp-content/uploads/2017/11/How-to-cook-jollof-rice-in-the-oven-750x500.jpg",
                              "price": "1500.00",
                              "title": "Jollof Rice, Beef and Plantain",
                              "vegetarian": false,
                            }
                          }
                          orderMeal={null}
                          toggleModal={null}
                          type="menu"
                        />,
                        <MealCard
                          inBasket={false}
                          meal={
                            Object {
                              "description": "Sharwama contains no animal products, perfect for dieters",
                              "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
                              "imageUrl": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
                              "price": "1200.00",
                              "title": "Vegetable Sharwama and Guava Smoothie",
                              "vegetarian": true,
                            }
                          }
                          orderMeal={null}
                          toggleModal={null}
                          type="menu"
                        />,
                      ]
                    }
                    loadMore={[Function]}
                    metadata={Object {}}
                  >
                    <div
                      className="card-group meals-wrapper"
                      id="card-group"
                    >
                      <InfiniteLoader
                        height={null}
                        items={
                          Array [
                            <MealCard
                              inBasket={false}
                              meal={
                                Object {
                                  "description": "Jollof Rice, Beef and Plantain. 2 pieces of beef per plate",
                                  "id": "81211c24-51c0-46ec-b1e0-18db55880958",
                                  "imageUrl": "http://www.preciouscore.com/wp-content/uploads/2017/11/How-to-cook-jollof-rice-in-the-oven-750x500.jpg",
                                  "price": "1500.00",
                                  "title": "Jollof Rice, Beef and Plantain",
                                  "vegetarian": false,
                                }
                              }
                              orderMeal={null}
                              toggleModal={null}
                              type="menu"
                            />,
                            <MealCard
                              inBasket={false}
                              meal={
                                Object {
                                  "description": "Sharwama contains no animal products, perfect for dieters",
                                  "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
                                  "imageUrl": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
                                  "price": "1200.00",
                                  "title": "Vegetable Sharwama and Guava Smoothie",
                                  "vegetarian": true,
                                }
                              }
                              orderMeal={null}
                              toggleModal={null}
                              type="menu"
                            />,
                          ]
                        }
                        loadMore={[Function]}
                        metadata={Object {}}
                      >
                        <InfiniteScroll
                          dataLength={2}
                          disableBrowserPullToRefresh={true}
                          hasMore={false}
                          height={null}
                          loader={
                            <div
                              className="text-center"
                            >
                              <MiniPreloader />
                            </div>
                          }
                          next={[Function]}
                          pullDownToRefreshContent={
                            <h3>
                              Pull down to refresh
                            </h3>
                          }
                          pullDownToRefreshThreshold={100}
                          releaseToRefreshContent={
                            <h3>
                              Release to refresh
                            </h3>
                          }
                        >
                          <div
                            style={Object {}}
                          >
                            <div
                              className="infinite-scroll-component "
                              style={
                                Object {
                                  "WebkitOverflowScrolling": "touch",
                                  "height": "auto",
                                  "overflow": "auto",
                                }
                              }
                            >
                              <div
                                className="scroller"
                              >
                                <MealCard
                                  inBasket={false}
                                  key="81211c24-51c0-46ec-b1e0-18db55880958"
                                  meal={
                                    Object {
                                      "description": "Jollof Rice, Beef and Plantain. 2 pieces of beef per plate",
                                      "id": "81211c24-51c0-46ec-b1e0-18db55880958",
                                      "imageUrl": "http://www.preciouscore.com/wp-content/uploads/2017/11/How-to-cook-jollof-rice-in-the-oven-750x500.jpg",
                                      "price": "1500.00",
                                      "title": "Jollof Rice, Beef and Plantain",
                                      "vegetarian": false,
                                    }
                                  }
                                  orderMeal={null}
                                  toggleModal={null}
                                  type="menu"
                                >
                                  <div
                                    className="meal-card"
                                    id="81211c24-51c0-46ec-b1e0-18db55880958"
                                  >
                                    <div
                                      className="meal-card-header"
                                    >
                                      <img
                                        alt="meal"
                                        src="http://www.preciouscore.com/wp-content/uploads/2017/11/How-to-cook-jollof-rice-in-the-oven-750x500.jpg"
                                      />
                                      <div
                                        className="menu-card-title"
                                      >
                                        <p>
                                          Jollof Rice, Beef and Plantain
                                        </p>
                                      </div>
                                    </div>
                                    <div
                                      className="meal-card-body"
                                    >
                                      <div>
                                        <h3>
                                          ₦
                                          1500.00
                                        </h3>
                                        <p
                                          className="meal-description"
                                          data-tip="Jollof Rice, Beef and Plantain. 2 pieces of beef per plate"
                                        >
                                          Jollof Rice, Beef and Plantain. 2 pieces of beef per plate
                                        </p>
                                      </div>
                                    </div>
                                    <ReactTooltip
                                      className="tooltip"
                                      clickable={false}
                                      effect="solid"
                                      insecure={true}
                                      place="top"
                                      resizeHide={true}
                                      type="dark"
                                      wrapper="div"
                                    >
                                      <div
                                        className="__react_component_tooltip place-top type-dark"
                                        data-id="tooltip"
                                      />
                                    </ReactTooltip>
                                  </div>
                                </MealCard>
                                <MealCard
                                  inBasket={false}
                                  key="36d525d1-efc9-4b75-9999-3e3d8dc64ce3"
                                  meal={
                                    Object {
                                      "description": "Sharwama contains no animal products, perfect for dieters",
                                      "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
                                      "imageUrl": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
                                      "price": "1200.00",
                                      "title": "Vegetable Sharwama and Guava Smoothie",
                                      "vegetarian": true,
                                    }
                                  }
                                  orderMeal={null}
                                  toggleModal={null}
                                  type="menu"
                                >
                                  <div
                                    className="meal-card"
                                    id="36d525d1-efc9-4b75-9999-3e3d8dc64ce3"
                                  >
                                    <div
                                      className="meal-card-header"
                                    >
                                      <span
                                        className="veg-ribbon"
                                      >
                                        Vegetarian
                                      </span>
                                      <img
                                        alt="meal"
                                        src="https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350"
                                      />
                                      <div
                                        className="menu-card-title"
                                      >
                                        <p>
                                          Vegetable Sharwama and Guava Smoothie
                                        </p>
                                      </div>
                                    </div>
                                    <div
                                      className="meal-card-body"
                                    >
                                      <div>
                                        <h3>
                                          ₦
                                          1200.00
                                        </h3>
                                        <p
                                          className="meal-description"
                                          data-tip="Sharwama contains no animal products, perfect for dieters"
                                        >
                                          Sharwama contains no animal products, perfect for dieters
                                        </p>
                                      </div>
                                    </div>
                                    <ReactTooltip
                                      className="tooltip"
                                      clickable={false}
                                      effect="solid"
                                      insecure={true}
                                      place="top"
                                      resizeHide={true}
                                      type="dark"
                                      wrapper="div"
                                    >
                                      <div
                                        className="__react_component_tooltip place-top type-dark"
                                        data-id="tooltip"
                                      />
                                    </ReactTooltip>
                                  </div>
                                </MealCard>
                              </div>
                            </div>
                          </div>
                        </InfiniteScroll>
                      </InfiniteLoader>
                    </div>
                  </CardGroup>
                </div>
              </div>
            </div>
            <Connect(Modal)
              isFetching={false}
              logout={[Function]}
              showTime={false}
              type="menu"
              updateCurrentDate={[Function]}
              uploading={false}
              user={
                Object {
                  "address": "$, Herbert Macaulay Way, Yaba, Lagos",
                  "businessName": "Food Circle",
                  "email": "food@circle.com",
                  "firstname": "Dave",
                  "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
                  "lastname": "Smith",
                  "phoneNo": "08166557788",
                  "role": "caterer",
                }
              }
            >
              <Modal
                isFetching={false}
                logout={[Function]}
                open={false}
                showTime={false}
                toggleModal={[Function]}
                type={null}
                updateCurrentDate={[Function]}
                uploading={false}
                user={
                  Object {
                    "address": "$, Herbert Macaulay Way, Yaba, Lagos",
                    "businessName": "Food Circle",
                    "email": "food@circle.com",
                    "firstname": "Dave",
                    "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
                    "lastname": "Smith",
                    "phoneNo": "08166557788",
                    "role": "caterer",
                  }
                }
              />
            </Connect(Modal)>
            <Footer>
              <div
                className="footer"
              >
                <p>
                  © 2018. Iveren Shaguy. All images were gotten from the internet and are not mine.
                </p>
              </div>
            </Footer>
          </div>
        </View>
      </Connect(View)>
    </CatererMenu>
  </Connect(CatererMenu)>
</Provider>
`;

exports[`CatererMenu should render Menu component correctly and show Meal Cards 1`] = `
<Fragment>
  <Connect(View)
    showTime={false}
    type="menu"
    updateCurrentDate={[Function]}
  >
    <div
      className="content-wrapper meals menu-meals"
    >
      <div
        className="date d-none-md"
      >
        <h2>
          Thursday, 1st January 1970
           
          <span
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
             ▾
          </span>
        </h2>
        <DatePicker
          handleSelectDate={[Function]}
          screenSize={null}
        />
      </div>
      <div
        className="top"
      >
        <button
          className="btn btn-pri"
          id="menu-modal-btn"
          onClick={[Function]}
          type="button"
        >
          Set Menu
        </button>
      </div>
      <div
        className="page-heading"
      >
        <h2>
          Meal Items on This Day's Menu
        </h2>
        <hr />
      </div>
      <CardGroup
        items={
          Array [
            <MealCard
              inBasket={false}
              meal={
                Object {
                  "description": "Jollof Rice, Beef and Plantain. 2 pieces of beef per plate",
                  "id": "81211c24-51c0-46ec-b1e0-18db55880958",
                  "imageUrl": "http://www.preciouscore.com/wp-content/uploads/2017/11/How-to-cook-jollof-rice-in-the-oven-750x500.jpg",
                  "price": "1500.00",
                  "title": "Jollof Rice, Beef and Plantain",
                  "vegetarian": false,
                }
              }
              orderMeal={null}
              toggleModal={null}
              type="menu"
            />,
            <MealCard
              inBasket={false}
              meal={
                Object {
                  "description": "Sharwama contains no animal products, perfect for dieters",
                  "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
                  "imageUrl": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
                  "price": "1200.00",
                  "title": "Vegetable Sharwama and Guava Smoothie",
                  "vegetarian": true,
                }
              }
              orderMeal={null}
              toggleModal={null}
              type="menu"
            />,
          ]
        }
        loadMore={[Function]}
        metadata={
          Object {
            "next": "next",
            "pages": 2,
            "prev": "prev",
            "total": 5,
          }
        }
      />
    </div>
  </Connect(View)>
</Fragment>
`;

exports[`CatererMenu should render message when there are no meals on the menu 1`] = `
<Fragment>
  <Connect(View)
    showTime={false}
    type="menu"
    updateCurrentDate={[Function]}
  >
    <div
      className="content-wrapper meals menu-meals"
    >
      <div
        className="date d-none-md"
      >
        <h2>
          Thursday, 1st January 1970
           
          <span
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
             ▾
          </span>
        </h2>
        <DatePicker
          handleSelectDate={[Function]}
          screenSize={null}
        />
      </div>
      <div
        className="top"
      >
        <button
          className="btn btn-pri"
          id="menu-modal-btn"
          onClick={[Function]}
          type="button"
        >
          Set Menu
        </button>
      </div>
      <div
        className="page-heading"
      >
        <h2>
          Meal Items on This Day's Menu
        </h2>
        <hr />
      </div>
      <p
        className="text-center info"
      >
        There are no Meal Items on this Menu
      </p>
    </div>
  </Connect(View)>
</Fragment>
`;
